starknet_py/contract.py

```diff
@@ -415,6 +483,81 @@ async def from_address(
 
         return Contract(address=address, abi=abi, client=client)
 
+    @staticmethod
+    async def declare(
+        account: AccountClient,
+        compiled_contract: str,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> DeclareResult:
+        """
+        Declares a contract
+
+        :param account: An AccountClient used to sign and send declare transaction.
+        :param compiled_contract: String containing compiled contract.
+        :param max_fee: Max amount of Wei to be paid when executing transaction.
+        :param auto_estimate: Use automatic fee estimation (not recommended, as it may lead to high costs).
+        :return: DeclareResult instance
+        """
+        declare_tx = await account.sign_declare_transaction(
+            compiled_contract=compiled_contract,
+            max_fee=max_fee,
+            auto_estimate=auto_estimate,
+        )
+        res = await account.declare(transaction=declare_tx)
+
+        return DeclareResult(
+            hash=res.transaction_hash,
+            _client=account.client,
+            class_hash=res.class_hash,
+            account=account,
+            compiled_contract=compiled_contract,
+        )
+
+    @staticmethod
+    async def deploy_contract(
+        class_hash: Hash,
+        account: AccountClient,
+        abi: List,
+        constructor_args: Optional[Union[List, Dict]] = None,
+        *,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> "DeployResult":
```

Comment:
What if we offered `prefunded_deploy` flag here to deploy using deploy account? 🤔

Reply:
Or a seperate method, `deploy_prefunded` for both Contract & DeclareResult?

Reply:
`deploy_prefunded` would be better. However, I would make it in a separate PR

Reply:
Let's have it as a separate method, it goes agains "simplified" to have many possible flows in single method.

Reply:
https://github.com/software-mansion/starknet.py/issues/515